app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -100,13 +100,17 @@ def queue_name_for_ems(ems)
       ems.queue_name
     end
 
-    def ems_id_from_queue_name(queue_name)
+    def parse_ems_id(queue_name)
       return nil if queue_name.blank?
       name, id = queue_name.split("_")
       return nil unless name == "ems"
       id.to_i
     end
 
+    def ems_id_from_queue_name(queue_name)
+      queue_name.kind_of?(Array) ? queue_name.collect { |q| parse_ems_id(q) }.flatten : parse_ems_id(queue_name)
```

Comment:
@juliancheal might `Array(queue_name).collect ...` make this simpler?  Would just be one case then

Reply:
So you mean, if `queue_name` isn't an Array turn in into one anyway?

Reply:
Yeah actually now that I think of it I like this better since 90% of the time it'll be not an array :)

Reply:
> This looks good to me, @juliancheal is there anything else that this depends on or is it okay to merge?

This one needs merging, https://github.com/ManageIQ/manageiq-providers-amazon/pull/280 but I don't know which one first tbh.

Reply:
right, we will need only IN query for it, but I think we can do that. Seems like PG is smart enough to convert IN with 1 element array into = (look at the index cond below, same applies for seq. scan)

```
   (1.0ms)  EXPLAIN ANALYZE SELECT "container_images".* FROM "container_images" WHERE (id IN (1710695))
 => [["Index Scan using container_images_pkey on container_images  (cost=0.43..8.45 rows=1 width=685) (actual time=0.026..0.027 rows=1 loops=1)"], ["  Index Cond: (id = 1710695)"], ["Planning time: 0.220 ms"], ["Execution time: 0.076 ms"]] 
2.3.0 :009 > ActiveRecord::Base.connection.query("EXPLAIN ANALYZE #{ContainerImage.where('id=1710695').to_sql}")
   (1.0ms)  EXPLAIN ANALYZE SELECT "container_images".* FROM "container_images" WHERE (id=1710695)
 => [["Index Scan using container_images_pkey on container_images  (cost=0.43..8.45 rows=1 width=685) (actual time=0.026..0.028 rows=1 loops=1)"], ["  Index Cond: (id = 1710695)"], ["Planning time: 0.147 ms"], ["Execution time: 0.076 ms"]]
```